protoarray/src/main/java/tech/pegasys/artemis/protoarray/ProtoArrayForkChoiceStrategy.java

```diff
@@ -139,13 +139,15 @@ private static void processBlockAtStartup(
         store.getBlockState(block.hash_tree_root()).getFinalized_checkpoint().getEpoch());
   }
 
-  void processAttestation(int validatorIndex, Bytes32 blockRoot, UnsignedLong targetEpoch) {
-    VoteTracker vote = votes.get(validatorIndex);
+  void processAttestation(MutableStore store, int validatorIndex, Bytes32 blockRoot, UnsignedLong targetEpoch) {
+    VoteTracker vote = store.getVote(validatorIndex).copy();
```

Comment:
This copy should be done inside `Store`.  There should be no way to modify things in the `Store` without going via the transaction.

Reply:
Done.